elasticsearch_dsl/faceted_search.py

```diff
@@ -168,6 +168,10 @@ def get_value_filter(self, filter_value):
         )
 
 
+def _date_interval_year(d):
+    return (d + timedelta(days=366)).replace(day=1)
```

Comment:
Let's change this to `return d.replace(year=d.year+1, day=(28 if d.month == 2 and d.day == 29 else d.day))` since years are linear unlike months.

We should add a test case for leap day handling as well (Basically see that February 29th, 2020 increments to February 28th, 2021)

Reply:
Done. Updated to ```return d.replace(year=d.year+1, day=(28 if d.month == 2 and d.day == 29 else d.day))```